### PR TITLE
infoschema:Migrate `dataForColumns` from package `infoschema` to `executor`

### LINKED ISSUE #15029: UCP: Migrate `dataForColumns` from package `infoschema` to `executor`
## Description

This is sub-task of https://github.com/pingcap/tidb/issues/14828

The package `infoschema` should be used to store meta-information for virtual tables only.
But there is some retrieval data logic of the virtual table, which messes the `infoschema` .
PR #14704 has begun to migrate them. You can reference #14704 to help you migrate the `dataForColumns`.

## Score 

* 50

## Mentor(s)

* @crazycs520 

## Recommended Skills

* Go

## Learning Materials

- PR: #14704

### PR DESCRIPTION
**Pull request by LindaSummer**
UCP #15029

### What problem does this PR solve?

fixes #15029

### What is changed and how it works?

What's Changed:

How it Works:

move ```dataForColumns``` from infoschema to executor

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: infoschema/tables.go [hunk 1]
```diff
@@ -49,8 +49,9 @@ const (
 	// TableSchemata is the string constant of infoschema table.
 	TableSchemata = "SCHEMATA"
 	// TableTables is the string constant of infoschema table.
-	TableTables           = "TABLES"
-	tableColumns          = "COLUMNS"
+	TableTables = "TABLES"
+	// TableColumns is the string constant of infoschema table
```
Comment (qw4990) [inline infoschema/tables.go]: This comment seems redundant？

**Comment by LindaSummer** [comment]
> Rest, LGTM. Thanks for your contribution @LindaSummer :D

thx for review, this is my first pr in opensource project :D.  I'm glad to make some contributions for tidb~ XD

File: infoschema/tables.go (hunk 1 above)
Comment (LindaSummer) [reply infoschema/tables.go]: it seems public constant before and after this line has comment like this, so add this comment, need i remove it?

**Comment by SunRunAway** [comment]
@reafans Please take another look.

**Pull request merged**